### PR TITLE
RAINCATCH-912 - sync client integration and docs

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Sync client interface improvements. 
Requires: https://github.com/feedhenry/fh-sync-js/pull/23

## Progress
- [x] Wraping interface
- [x] Unit tests
- [x] Documentation

### CODE REVIEW

File: client/datasync-client/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  cloudUrl: 'http://localhost:3000',
+  sync_frequency: 10,
+  storage_strategy: 'dom'
+};
+$fh.sync.init(options);
+```
+
+### Register your local data collections
+
+Sync introduces concept of `DataSets`. DataSets are form of local data collections
```
Comment (wtrocki) [inline client/datasync-client/README.md]: @austincunningham

File: client/datasync-client/integration/integration.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const metaData = {};
+
+const task: any = {
+  name: 'test task',
+  status: 'finished'
+};
+
+$fh.sync.manage(datasetId, options, queryParams, metaData, function() {
+  $fh.sync.doCreate(datasetId, task, function(data) {
+    // tslint:disable-next-line:no-console
```
Comment (austincunningham) [inline client/datasync-client/integration/integration.ts:25]: does this comment need to be here see it repeated throughout your code

File: client/datasync-client/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+$fh.sync.manage(datasetId, options, queryParams, metaData, function() {
+    // callback code
+});
+```
+
+### Perform operations on datasets
+
+Sync provides set of methods to perform create, read, update and delete operations on datasets.
+
```
Comment (austincunningham) [inline client/datasync-client/README.md:60]: maybe stick 'list' in here for full CRUDL

File: client/datasync-client/integration/integration.ts (hunk 2 above)
Comment (wtrocki) [reply client/datasync-client/integration/integration.ts:25]: This is special comment that disables tslint rule: 
https://palantir.github.io/tslint/usage/rule-flags/

That's integration file so we actually want to use console.log

**Comment by wtrocki** [comment]
@JameelB @paolobueno - Can you help me to verify that change? This will block you with WFM work. Additional requirements and helpers can be incorporated later (possibly directly to sync rather than to our module)

**Review by JameelB — APPROVED** [review]

**Review by austincunningham — APPROVED** [review]
verified

**Pull request merged**